lib/page-object/section.rb

```diff
@@ -0,0 +1,40 @@
+module PageObject
+  class Section < Elements::Element
+    extend Accessors
+    attr_reader :platform
+    #Given that page sections implement their own behaviors,
+    #we don't want misses to fall through to the watir element
+    undef_method :method_missing
+
+    private
+    #there is currently no better way to handle the dual nature of sections being like elements and pages
+    def call_block(&block)
```

Comment:
This was the hackiest part of this code. call_block is implemented in the PageObject module, and is needed for the element accessors. But the PageObject module can't be included at the same time as inheriting Element, because element and PageObject have conflicting initialization. I could split this method out into its own module and include that in both places, but I wanted to get feedback before moving forward with that.